### PR TITLE
Adds spinner to "Add Links" Modal.

### LINKED ISSUE #3494: [Firefox] OSF Spinner does not spin on Files pages
## Steps
1. Using the Firefox browser, go to a project you have read+write permissions
2. Go to the Files Tab
3. Look for the OSF spinner as the page loads, refresh the page if needed
4. Click on a file to go to the files detail page
5. Look for the OSF spinner as the page loads where the file should render, refresh the page if needed

## Expected
I would expect the OSF spinner to spin

## Actual
On these pages, the spinner flashes or is constant, but does not spin

### LINKED ISSUE #3611: "Add Links" modal: Add spinner when search results are loading
## Steps

1. Log in to osf.io
2. Navigate to one of your projects.
3. In the components widget, click on "Add Links."
4. Search all projects for "Psych".
5. From page 1 of the results, click to the last page of the results.
6. Notice the behavior.

## Expected Behavior

I expect that while the page loads, I will see a spinner, as when logs are loading:

loading_logs

## Actual Behavior

There are 28 pages of results in this example, and clicking between pages (especially from page 1 to page 28) takes awhile. While the load occurs, nothing appears to be happening. With more pages of results, the lack of a spinner informing the user what is going on becomes more noticeable. Blue highlighting of the page number doesn't occur until the load has finished, so the user doesn't know for sure if the request is being processed. Adding a spinner will enhance the user experience.

### PR DESCRIPTION
**Pull request by Johnetordoff**
<h1>Purpose</h1>
Add spinner to "Add Link" modal. Closes #3611.
<h1>Changes</h1>
Small changes to modal_add_pointers.mako and pointers.js.
<h1>Side Effects</h1>
None that I know of.

### CODE REVIEW

File: website/static/js/pointers.js [hunk 1]
```diff
@@ -69,9 +75,13 @@ var AddPointerViewModel = oop.extend(Paginator, {
                 self.currentPage(result.page);
                 self.numberOfPages(result.pages);
                 self.addNewPaginators();
+                self.loadingResults(false);
```
Comment (sloria) [inline website/static/js/pointers.js]: This should probably go in the `always` callback; we want to hide the loading indicator whether or not we get a success or a failure response.

File: website/static/js/pointers.js [hunk 2]
```diff
@@ -54,6 +55,11 @@ var AddPointerViewModel = oop.extend(Paginator, {
         self.searchWarningMsg('');
 
         if (self.query()) {
+            var timeout = setTimeout(function() {
+            self.results([]); // clears page for spinner
```
Comment (sloria) [inline website/static/js/pointers.js]: Fix indentation.

File: website/templates/project/modal_add_pointer.mako [hunk 3]
```diff
@@ -49,6 +50,13 @@
                             </tbody>
                         </table>
                         <div class='help-block'>
+                            <div data-bind='if: loadingResults'>
+                                <div class="spinner-loading-wrapper">
+                                    <div class="logo-spin text-center"><img src="/static/img/logo_spin.png" alt="loader"> </div>
+                                    <p class="m-t-sm fg-load-message"> Loading files...  </p>
```
Comment (sloria) [inline website/templates/project/modal_add_pointer.mako]: Should this be "Loading results. . ."?

**Comment by sloria** [comment]
Pass finished. :point_up:

File: website/static/js/pointers.js [hunk 4]
```diff
@@ -69,9 +74,14 @@ var AddPointerViewModel = oop.extend(Paginator, {
                 self.currentPage(result.page);
                 self.numberOfPages(result.pages);
                 self.addNewPaginators();
+                self.loadingResults(false);
             }).fail(function(xhr) {
                     self.searchWarningMsg(xhr.responseJSON && xhr.responseJSON.message_long);
+            }).always( function (){
+                clearTimeout(timeout); // clear timeout function
+                this.loadingResults = ko.observable(false);
```
Comment (sloria) [inline website/static/js/pointers.js]: `this` is not the viewmodel here, since you are within a function. Use `self`.

**Comment by sloria** [comment]
I'm seeing an error: 

```js
Uncaught ReferenceError: Unable to process binding "if: function (){return loadingResults }"
Message: loadingResults is not defined
```

**Comment by sloria** [comment]
Pass finished. This doesn't seem to be working as is. Please debug, and I'll do another pass.

**Comment by Johnetordoff** [comment]
@sloria this should be good to go now.

**Pull request merged**